app/src/main/java/com/epmedu/animeal/component/IntentGpsSettingsProvider.kt

```diff
@@ -0,0 +1,48 @@
+package com.epmedu.animeal.component
+
+import android.content.Context
+import android.content.Intent
+import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
+import android.location.GnssStatus
+import android.location.LocationManager
+import android.os.Handler
+import android.os.Looper
+import android.provider.Settings
+import com.epmedu.animeal.common.component.GpsSettingsProvider
+import kotlinx.coroutines.channels.ProducerScope
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.callbackFlow
+
+/** Allows to enable/disable gps settings via intent call to the system */
+class IntentGpsSettingsProvider(private val androidApplicationContext: Context) : GpsSettingsProvider {
+
+    override val isGpsSettingsEnabled: Boolean
+        get() = androidApplicationContext.getLocationManager()?.isProviderEnabled(LocationManager.GPS_PROVIDER) ?: false
+
+    // FLAG_ACTIVITY_NEW_TASK is required when we're starting activity from app context
+    override fun changeGpsSettings() = androidApplicationContext.startActivity(
```

Comment:
I suggest move this logic out of this class and open settings screen directly from Composable function. You can make some extension for this

Reply:
It causes on click event. Click event isn't a composable function at all, so we should bring scope into this function or call a viewmodel. Do we really need to write such navigation logic in ui?

Reply:
well, lets leave in provider now